### PR TITLE
Add unit tests for linears.py

### PR DESCRIPTION
**Pull request by bvandermoon**
# Description
This PR adds unit tests for `src/maxtext/layers/linears.py`, covering:
- Utility functions: `normalize_axes`, `canonicalize_tuple`, `_convert_to_activation_function`.
- `DenseGeneral` layer: Basic call, bias usage, and multiple axes.
- `MlpBlock` layer: Basic call and fused MLP configuration.
These tests help ensure the correctness of these core layers and improve test coverage.

# Tests
I tested this change by running the new unit tests on a TPU VM.
Command: `pytest tests/unit/linears_test.py`
Result: `8 passed`

# Checklist
Before submitting this PR, please make sure (put X in square brackets):
- [X] I have performed a self-review of my code. For an optional AI review, add the `gemini-review` label.
- [X] I have necessary comments in my code, particularly in hard-to-understand areas.
- [X] I have run end-to-end tests tests and provided workload links above if applicable.
- [X] I have made or will make corresponding changes to the doc if needed, including adding new documentation pages to the relevant Table of Contents (toctree directive) as explained in [our documentation](https://maxtext.readthedocs.io/en/latest/development.html#adding-new-documentation-files).

### CODE REVIEW

**Review by igorts-git — APPROVED** [review]

File: tests/unit/linears_test.py [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+  def test_multiple_axes(self):
+    batch_size = 2
+    seq_len = 3
+    in_features = 4
+    out_features = 8
+
+    layer = linears.DenseGeneral(
+        in_features_shape=in_features,
+        out_features_shape=out_features,
+        axis=-1,
```
Comment (NuojCheng) [inline tests/unit/linears_test.py]: `axis=-1` is the default value. Maybe test when `axis=0` or `axis=1`?

File: tests/unit/linears_test.py [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+    layer = linears.MlpBlock(
+        config=self.cfg,
+        mesh=self.mesh,
+        in_features=in_features,
+        intermediate_dim=intermediate_dim,
+        rngs=self.rngs,
+    )
+
+    inputs = jnp.ones((batch_size, seq_len, in_features))
+    outputs = layer(inputs)
```
Comment (NuojCheng) [inline tests/unit/linears_test.py:167]: for mlp, maybe add weight shape checks? e.g. 
assert layers.variables['params']['wi']['kernel'].shape == (in_features, intermediate_dim)

**Review by NuojCheng — APPROVED** [review]

File: tests/unit/linears_test.py (hunk 1 above)
Comment (bvandermoon) [reply tests/unit/linears_test.py]: Done, thanks @NuojCheng

File: tests/unit/linears_test.py (hunk 2 above)
Comment (bvandermoon) [reply tests/unit/linears_test.py:167]: Done, thanks @NuojCheng
